stingray/fourier.py

```diff
@@ -0,0 +1,1286 @@
+        dRe = dIm = dG = np.sqrt(PoN * (Ps - frac))
+        # Eq. 19
+        dphi = np.sqrt(PoN * (Ps / (Gsq - bsq) - 1 / (Pr - Prnoise)))
+
+    else:
+        PrPs = Pr * Ps
+        dRe = np.sqrt((PrPs + C.real ** 2 - C.imag ** 2) / twoN)
+        dIm = np.sqrt((PrPs - C.real ** 2 + C.imag ** 2) / twoN)
+        gsq = raw_coherence(C, Ps, Pr, Psnoise, Prnoise, Nave)
+        dphi = np.sqrt((1 - gsq) / (2 * gsq ** 2 * Nave))
+        dG = np.sqrt(PrPs / Nave)
+
+    return dRe, dIm, dphi, dG
+
+
+def cross_to_covariance(C, Pr, Prnoise, delta_nu):
+    """Convert a cross spectrum into a covariance.
+     Parameters
+    ----------
+    C : complex `np.array`
+        cross spectrum
+    Pr : float `np.array`
+        reference-band periodogram
+    Prnoise : float
+        Poisson noise level of the reference-band periodogram
+    delta_nu : float or `np.array`
+        spectral resolution. Can be a float, or an array if the spectral
+        resolution is not constant throughout the periodograms
+
+    """
+    return C * np.sqrt(delta_nu / (Pr - Prnoise))
+
+
+def _which_segment_idx_fun(counts=None, dt=None):
+    # Make function interface equal (counts gets ignored)
+    if counts is None:
+        return generate_indices_of_segment_boundaries_unbinned
+
+    def fun(*args, **kwargs):
+        return generate_indices_of_segment_boundaries_binned(*args, dt=dt, **kwargs)
+    return fun
+
+
+def get_total_ctrate(times, gti, segment_size, counts=None):
+    """Calculate the average count rate during the observation.
+
+    This function finds the same segments that the PDS will use and
+    returns the mean count rate.
+    If ``counts`` is ``None``, the input times are interpreted as events.
+    Otherwise, the number of events is taken from ``counts``
+
+    Parameters
+    ----------
+    times : float `np.array`
+        Array of times
+    gti : [[gti00, gti01], [gti10, gti11], ...]
+        good time intervals
+    segment_size : float
+        length of segments
+
+    Other parameters
+    ----------------
+    counts : float `np.array`, default None
+        Array of counts per bin
+
+    Examples
+    --------
+    >>> times = np.sort(np.random.uniform(0, 1000, 1000))
+    >>> gti = np.asarray([[0, 1000]])
+    >>> counts, _ = np.histogram(times, bins=np.linspace(0, 1000, 11))
+    >>> bin_times = np.arange(50, 1000, 100)
+    >>> get_total_ctrate(bin_times, gti, 1000, counts=counts)
+    1.0
+    >>> get_total_ctrate(times, gti, 1000)
+    1.0
+    """
+    Nph = 0
+    Nintvs = 0
+    func = _which_segment_idx_fun(counts)
+
+    for _, _, idx0, idx1 in func(times, gti, segment_size):
+        if counts is None:
+            Nph += idx1 - idx0
+        else:
+            Nph += np.sum(counts[idx0:idx1])
+        Nintvs += 1
+
+    return Nph / (Nintvs * segment_size)
+
+
+def get_flux_iterable_from_segments(times, gti, segment_size, N=None, counts=None, errors=None):
+    """Get fluxes from different segments of the observation.
```

Comment:
I'm not really sure what you mean here by "flux"? Do we not usually work in counts everywhere?

Reply:
In principle, these functions might be used with arrays of count rates, or background subtracted. This is why I just say "flux" here. There might be something wrong in the normalizations maybe...

Reply:
I totally agree with having a consistent language in this module, and possibly throughout the code base. In fourier.py, however, I think that the functions might be more general than the rest of the code base. Here, we might accept normalizations that are not of interest for the current periodograms, or light curves in units that are not yet available in Lightcurve. Indeed, much of the machinery does not need the light curves to be in counts. Following this reasoning ("we can be more general here"), I used consistently `fluxes` in this module. Only exception get_average_ctrate, where indeed the input needs to be in units of counts.